src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -84,6 +96,38 @@ public String updateUserEmail(UserChangeEmailDto userChangeEmailDto, User user)
         return getMessageSource("changed.user.email");
     }
 
+    public String deleteUserFirst(User user) {
```

Comment:
Wrong name - you don't delete first user but you delete user by it's request. My suggestion: change it to **selfDeleteRequest**